packages/app/src/specs/InlineSpecList.cy.tsx

```diff
@@ -222,5 +222,30 @@ describe('InlineSpecList', () => {
       mountInlineSpecList({ experimentalRunAllSpecs: true, specFilter: 'B' })
       hoverRunAllSpecs('src/components', 1)
     })
+
+    it('allows keyboard interactions to run spec groups without toggling sections', () => {
+      mountInlineSpecList({ experimentalRunAllSpecs: true })
+
+      const mutationStub = cy.stub().as('mutationStub')
+
+      cy.stubMutationResolver(RunAllSpecsDocument, () => {
+        mutationStub()
+      })
+
+      cy.findAllByTestId('spec-file-item').should('have.length', 4)
+      cy.findAllByTestId('run-all-specs-button').eq(0)
+      .click()
+      .type(' ')
+
+      // make sure typing didn't change displayed items
+      cy.findAllByTestId('spec-file-item').should('have.length', 4)
+      cy.findAllByTestId('run-all-specs-button').eq(1)
+      .focus()
+      .type('{enter}')
+
+      // make sure typing didn't change displayed items
+      cy.findAllByTestId('spec-file-item').should('have.length', 4)
+      cy.get('@mutationStub').should('have.been.calledThrice')
```

Comment:
Same here.

I think we should delete one of the two tests you've added? Maybe `InlineRunAllSpecs.cy.tsx` - they test *exactly* the same thing, but this one covers a bit more.

Reply:
The `InlineRunAllSpecs` component is responsible for managing its own keyboard behavior, so ideally it should be tested there. The component was already working correctly, so in this PR I'm just adding a case that I thought was missing from the original.

`InlineRunAllSpecs` is used by 2 other components, one of which happened to break keyboard behavior because it's nesting controls in a non-standard way. So the test here is more about adding coverage for the bug we're fixing - something that fails on the old code and passes on this branch. 

But I think it's confusing if this is the only coverage of the behavior itself, we should be able to delete this spec without losing coverage on other components that have their own component tests.

I'll add a note to this test that we can remove it at a future point when we are no longer nesting a button within a button. Actually I'll make the issue for this first and then comment with the issue number - we can improve both specs lists by using a similar pattern to our card component, where the target that receives keyboard focus is a real button inside the larger element, even though the focus style and click listeners apply to the whole element. Then we can have internal controls in a row that aren't nested inside other controls, and it fixes a big problem where the whole row is announced as a label.

A bigger change than I wanted to make in this PR (keyboard behavior is still a win), but would solve the problems with the current implementation.

Reply:
That said, I did simplified the `InlineRunAllSpecs` to not repeat the keyboard stuff specifically. Since we already confirm it's a button element, the behavior on space and enter keys is free since a click is emitted by them, so we keep the keyboard specific stuff in this test, where it was broken, and avoid duplicating in the other component test. We still want to test the event emitted by that component.

Also made https://github.com/cypress-io/cypress/issues/24966 and added a TODO comment for future cleanup.